lib/types/object.js

```diff
@@ -50,8 +51,25 @@ module.exports = {
 
     // reject object-like values that have their own strong-remoting type
 
-    if (Array.isArray(value))
-      return errorNotAnObject();
+    if (Array.isArray(value)) {
+      // TODO: @davidcheung, remove this flag and support [array or Object]
+      // see strong-remoting/issues/360 for details
+      // allowArray flag is to handle persistedModels uses
+      // array of Objects to batch create, which was supported in 2.x
+      if (!options.allowArray) {
+        return errorNotAnObject();
+      } else {
+        var err = null;
+        value.forEach(function(item) {
+          if (typeof item !== 'object' || Array.isArray(item) ||
+          item === null) {
+            err = errorArrayItemsNotAnObject();
+            return;
+          }
+        });
```

Comment:
I think this may be a bit cleaner:

```js
var hasInvalidItems = value.some(function(item) {
  return typeof item !== 'object' || Array.isArray(item) || item === null;
});

return hasInvalidItems ? errorArrayItemsNotAnObject() : null;
```

How about Dates as the array items, I think we should reject them too. Can we perhaps reuse `validate` inside the forEach/some callback?

```js
var self = this;
var hasInvalidItems = value.some(function(item) {
  return !self.validate(item/* no options (?) */);
});
return hasInvalidItems ? errorArrayItemsNotAnObject() : null;
```